File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book.sort() # 정렬하는 이유는 시작하는 숫자가 동일해야 의미가 있기때문
    for i in range(len(phone_book) - 1):
        if len(phone_book[i]) < len(phone_book[i+1]):
            if phone_book[i] == phone_book[i+1][:len(phone_book[i])]:
                answer = False
                break
    return answer

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var result = true;
    phone_book.sort() // 정렬
    for (let i = 0; i < phone_book.length - 1; i++){
        if(phone_book[i].length < phone_book[i+1].length){
            if(phone_book[i] == phone_book[i+1].slice(0, phone_book[i].length)){
                result = false
                break
            }
        }
    }
    return result;
}